File: cust_field/cust_field/customer/email_custom.js
frappe.ui.form.on('Customer', {
	refresh: function(frm) {
       //frm.set_value("customer_name","AAAAA");
	//   frm.set_value("account_name", data.account_name);
	},
	 
	email: function(frm){
		var email_id = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
		console.log("============email function called")
        console.log(email_id.test(frm.doc.email));
		if (email_id.test(frm.doc.email) == false)
			{
				frm.set_value(`email`,``)
				frappe.throw("Incorrect Email")
			}
	},
	onload_post_render: function(frm) {
		 
        console.log("onload_post_render called...................");
    },
    validate:function(frm){
    	console.log("Validate^^^^^^^^^^^^^^...................");
    	frm.set_df_property("email","read_only",1);
    }   

});

File: cust_field/cust_field/customer/email_custom.py
from __future__ import unicode_literals
import frappe


def email(doc,method=None):
	print("#########################");
	print("@@@@@@@@@@@@@@@@@@@@@@@@@");
	frappe.msgprint('Successfully............')



def cre_custm(doc,method=None):
   print('++++++++Validation called++++++++++++++++++++++++++++++++++++')
   new_cus = frappe.new_doc("User")
   new_cus.email=doc.email
   print("Printing new user email")
   print(new_cus.email)
   print(doc.email)
   full_name = doc.customer_name
   full_name = full_name.split(' ')
   print(full_name)
   if(len(full_name)>1):
   		new_cus.first_name = full_name[0]
   		new_cus.last_name = full_name[1]
   else:
   		new_cus.first_name = full_name[0]
   new_cus.insert()
   frappe.db.commit()
   print("Full name of customer    =>  "+doc.customer_name)
   print('************************** '+doc.email+ '   **************************')	
   frappe.msgprint('Successfully Created............')



